### ISSUE TITLE
storefront_ui docker doesn't start up

### ISSUE THREAD

**Issue by marcodisarno** [kodbruket/vsf-payment-klarna#27]
I tried including the KlarnaCheckout in my project but I'm stuck at a certain point, so I thought let's give your example a good look.

But when trying to `docker-compose up`, the *storefront_ui* container crashes after successfully building. Here's its last words:

```
...
storefront-ui_1   | <s> [webpack.Progress] 100% 
storefront-ui_1   | 
storefront-ui_1   | Hash: 9f5d9ecbd18cb1aa32f1
storefront-ui_1   | Version: webpack 4.39.0
storefront-ui_1   | Time: 122411ms
storefront-ui_1   | Built at: 08/02/2019 10:41:43 AM
storefront-ui_1   |  46 assets
storefront-ui_1   | Entrypoint app = manifest.9f5d9ecbd18cb1aa32f1.js vendor.9f5d9ecbd18cb1aa32f1.js app.9f5d9ecbd18cb1aa32f1.js
storefront-ui_1   | 
storefront-ui_1   | ERROR in ./src/modules/payment-klarna/hooks/beforeRegistration.ts
storefront-ui_1   | Module not found: Error: Can't resolve '@vue-storefront/core/lib/router-manager' in '/var/www/src/modules/payment-klarna/hooks'
storefront-ui_1   |  @ ./src/modules/payment-klarna/hooks/beforeRegistration.ts 2:0-72 19:4-17 21:4-17
storefront-ui_1   |  @ ./src/modules/payment-klarna/index.ts
storefront-ui_1   |  @ ./src/modules/index.ts
storefront-ui_1   |  @ ./core/modules-entry.ts
storefront-ui_1   |  @ ./core/app.ts
storefront-ui_1   |  @ ./core/client-entry.ts
storefront-ui_1   |  @ multi babel-polyfill ./core/client-entry.ts
storefront-ui_1   | 
storefront-ui_1   | ERROR in /var/www/src/modules/payment-klarna/hooks/beforeRegistration.ts
storefront-ui_1   | [tsl] ERROR in /var/www/src/modules/payment-klarna/hooks/beforeRegistration.ts(2,31)
storefront-ui_1   |       TS2307: Cannot find module '@vue-storefront/core/lib/router-manager'.
storefront-ui_1   | 
storefront-ui_1   | ERROR in /var/www/src/modules/payment-klarna/store/getters.ts
storefront-ui_1   | [tsl] ERROR in /var/www/src/modules/payment-klarna/store/getters.ts(12,37)
storefront-ui_1   |       TS2345: Argument of type '{ name: string; fullPath: any; params: { parentSku: any; slug: any; childSku: any; }; }' is not assignable to parameter of type 'string | Route'.
storefront-ui_1   |   Type '{ name: string; fullPath: any; params: { parentSku: any; slug: any; childSku: any; }; }' is missing the following properties from type 'Route': path, hash, query, matched
storefront-ui_1   | Child html-webpack-plugin for "index.amp.html":
storefront-ui_1   |      1 asset
storefront-ui_1   |     Entrypoint undefined = index.amp.html
storefront-ui_1   | Child html-webpack-plugin for "index.basic.html":
storefront-ui_1   |      1 asset
storefront-ui_1   |     Entrypoint undefined = index.basic.html
storefront-ui_1   | Child html-webpack-plugin for "index.html":
storefront-ui_1   |      1 asset
storefront-ui_1   |     Entrypoint undefined = index.html
storefront-ui_1   | Child html-webpack-plugin for "index.minimal.html":
storefront-ui_1   |      1 asset
storefront-ui_1   |     Entrypoint undefined = index.minimal.html
storefront-ui_1   | error Command failed with exit code 2.
storefront-ui_1   | info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
vsf-payment-klarna_storefront-ui_1 exited with code 1
```

Any ideas?

PS: Some env info:
- tried master@ea1bd082791e9ec6970b3a6fc9cd8977210f04fb
- docker-compose version 1.22.0, build f46880fe
- Docker version 18.06.1-ce, build e68fc7a
- OS: Debian GNU/Linux 8.7 (jessie)

**Comment by benjick** [comment]
Hey, can you try this?

```
cd vsf-storefront
git pull origin master
git checkout ea5dcb4217f0c03123eaa898f62a43a5d1455945
```

It's probably because we've been updating this plugin from our other projects, sorry about that. Please send a PR if it's working

**Comment by marcodisarno** [comment]
Thanks for the quick reply. Tried it, now `.babelrc` seems to be missing in the docker

```
...
Starting vsf-payment-klarna_storefront-ui_1 ... error
...
ERROR: for storefront-ui  Cannot start service storefront-ui: OCI runtime create failed:
  container_linux.go:348: starting container process caused "process_linux.go:402:
    container init caused \"rootfs_linux.go:58:
      mounting "/home/developer/workspace/vsf-payment-klarna/vue-storefront/.babelrc"
      to rootfs "/var/lib/docker/aufs/mnt/bcc5adc910af2801bb559be55ea5be6855cb8bcb8a11cacb8087502de21c5496"
      at "/var/lib/docker/aufs/mnt/bcc5adc910af2801bb559be55ea5be6855cb8bcb8a11cacb8087502de21c5496/var/www/.babelrc"
      caused "not a directory": unknown:

Are you trying to mount a directory onto a file (or vice-versa)? Check if the specified host path exists and is the expected type

ERROR: Encountered errors while bringing up the project.
```

**Comment by benjick** [comment]
@marcodisarno woops

In `docker-compose.override.yml` add `- './vue-storefront-api/babel.config.js:/var/www/babel.config.js'` under services.api.volumes

and under `services.ui.volumes` replace `- './vue-storefront/.babelrc:/var/www/.babelrc'` with `- './vue-storefront/babel.config.js:/var/www/babel.config.js'`

**Comment by marcodisarno** [comment]
@benjick the good news is it builds and runs now, thanks for that.

There was a `graphql` dependency problem with `storefront-api` that I solved [like these ppl on the VSF forums did](https://forum.vuestorefront.io/t/dependency-issue-typeerror-graphql-1-getdescription-is-not-a-function/502).

`./vue-storefront-api/package.json`
```
"graphql": "^0.10.1" -> "^0.13.1"
```

**Note**: had to do it manually, though, because trying to solve this **the clean way** by updating submodule `vue-storefront-api` to [0ec68cb2](https://github.com/DivanteLtd/vue-storefront-api/commit/0ec68cb2c333506ce5e1f7c81ce7e0cfdd04dfd2) (where the issue is officially solved) resulted in: 

```
Unexpected Identifier: import http from 'http'
```

![image](https://user-images.githubusercontent.com/14130041/62470241-4a3eff80-b79a-11e9-874c-cdf4c156880b.png)

And the bad news is that now I'm stuck with a permanent loadingSpinner on the `KlarnaCheckout` like the reporter of issue #3

So all of this did not help me integrating vsf-payment-klarna 😭 


Still, if you give me permission I will push my (dirty) solution.

**Comment by marcodisarno** [comment]
Had to apply the following tweaks to make it actually load:
- the `merchant_urls` were missing in the order (getters/order)
- added `?cartId={cartId}` to API endpoint query (config.klarna.endpoint)

![image](https://user-images.githubusercontent.com/14130041/62693665-36caa900-b9d3-11e9-8335-1ccf216e96b3.png)

So I tried taking away the shipping options by setting `showShippingOptions: false`. Still resulted in the same problem:

![image](https://user-images.githubusercontent.com/14130041/62696832-8dd37c80-b9d9-11e9-8154-ffae3e3a0d61.png)

Searching the web brought forward a [problem the WooCommerce (https://github.com/krokedil/klarna-checkout-for-woocommerce/issues/195) people have]:

They claim that the checkout [won't work from a closed/local environment](https://github.com/krokedil/klarna-checkout-for-woocommerce/issues/195#issuecomment-493086842), because Klarna sends a callback to the store that needs to be responded to correctly.

Could that be true?

**Comment by gabwik** [comment]
Hi, we're testing on public and local environments. The message you see you see is not related to KLarna callback.

**Comment by benjick** [comment]
@marcodisarno please look into https://ngrok.com/, it might help you

**Comment by benjick** [comment]
Will be fixed in #65

**Comment by damirJa** [comment]
Hi, I had a problem with the building stage, because of babel-eslint. If you will be faced with, use that solution - https://github.com/babel/babel-eslint/issues/681#issuecomment-420663038

**Comment by benjick** [comment]
Hey. This is fixed now (#65) and this project is in much better shape. Sorry about the delay
